core/src/main/java/com/oath/oak/OakAttachedReadBuffer.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2018 Oath Inc.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.oath.oak;
+
+import sun.nio.ch.DirectBuffer;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+class OakAttachedReadBuffer implements OakReadBuffer, OakUnsafeDirectBuffer {
+
+    protected ByteBuffer bb;
+    protected int dataPos;
+    protected int dataLength;
+
+    OakAttachedReadBuffer(Slice s, ValueUtils valueOperator) {
+        bb = s.getByteBuffer();
+        dataPos = bb.position() + valueOperator.getHeaderSize();
```

Comment:
According to this line it can not be the "big block" ByteBuffer... So should be part of the thread array buffers. But then it is unsafe and may be wrong to keep the ByteBuffer predefined...

Reply:
Unsafe in what matter? It is an attached (scoped) buffer.

Reply:
I understand now

Reply:
What does valueOperator.getHeaderSize() return for keys?

Reply:
And one more thing, why to provide the value operator just in order to get header size from it why not to provide the header size instead?

Reply:
This was the design as it was before this PR. I fixed it now.